core/src/main/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizer.java

```diff
@@ -39,8 +41,21 @@ public LogicalPlanOptimizer(List<Rule<?>> rules) {
    */
   public static LogicalPlanOptimizer create() {
     return new LogicalPlanOptimizer(Arrays.asList(
+        /*
+         * Phase 1: Transformations that rely on relational algebra equivalence
+         */
         new MergeFilterAndFilter(),
-        new PushFilterUnderSort()));
+        new PushFilterUnderSort(),
+        /*
+         * Phase 2: Transformations that rely on data source push down capability
+         */
+        new CreateTableScanBuilder(),
```

Comment:
Does the rule order matters?  New rules should be append after phase 2?

Reply:
AFAIK, phase 1 has to run before phase 2. In phase 2, `CreateTableScanBuilder` rule has to precede other push down rules. However, for new rules, we may need to analyze dependency case by case.